components/EditTagsModal.tsx

```diff
@@ -0,0 +1,164 @@
+import React from 'react';
+import { gql, useMutation, useQuery } from '@apollo/client';
+import { Form, Formik } from 'formik';
+import { FormattedMessage } from 'react-intl';
+
+import { IGNORED_TAGS } from '../lib/constants/collectives';
+import { API_V2_CONTEXT, gqlV1 } from '../lib/graphql/helpers';
+import { Collective } from '../lib/graphql/types/v2/graphql';
+
+import CollectiveTagsInput from './CollectiveTagsInput';
+import { Flex } from './Grid';
+import MessageBox from './MessageBox';
+import StyledButton from './StyledButton';
+import StyledInputFormikField from './StyledInputFormikField';
+import StyledModal, { ModalBody, ModalFooter, ModalHeader } from './StyledModal';
+import { TOAST_TYPE, useToasts } from './ToastProvider';
+
+const editTagsMutation = gqlV1`
+  mutation editCollective(
+    $collective: CollectiveInputType!
+  ) {
+    editCollective(
+      collective: $collective
+    ) {
+      id
+      tags
+    }
+  }
+`;
+
+const tagStatsQuery = gql`
+  query tagStats($host: AccountReferenceInput) {
+    tagStats(host: $host, limit: 5) {
+      nodes {
+        id
+        tag
+      }
+    }
+  }
+`;
+
+export type EditTagsModalProps = {
+  collective: Collective;
+  onClose: () => void;
+};
+
+export default function EditTagsModal({ collective, onClose }: EditTagsModalProps) {
+  const { addToast } = useToasts();
+  const [editTags, { loading }] = useMutation(editTagsMutation);
+
+  const { data: { tagStats } = { tagStats: null } } = useQuery(tagStatsQuery, {
+    variables: { ...(collective.host?.slug ? { host: { slug: collective.host.slug } } : {}) },
+    context: API_V2_CONTEXT,
+  });
+
+  const initialValues = {
+    tags: collective.tags,
+  };
+
+  const submit = async values => {
+    const { tags } = values;
+    try {
+      const variables = {
+        collective: {
+          id: collective.id,
+          tags: tags,
+        },
+      };
+
+      await editTags({ variables });
+    } catch (e) {
+      addToast({
+        type: TOAST_TYPE.ERROR,
+        message: (
+          <FormattedMessage
+            defaultMessage="Error submiting form: {error}"
+            values={{
+              error: e.message,
+            }}
+          />
+        ),
+      });
+      return;
+    }
+    addToast({
+      type: TOAST_TYPE.SUCCESS,
+      message: <FormattedMessage defaultMessage="Successfully updated tags" />,
+    });
+    handleClose();
+  };
+
+  const handleClose = () => {
+    onClose?.();
+  };
+
+  return (
+    <StyledModal flexGrow={1} maxWidth="500px" onClose={handleClose} trapFocus>
+      <Formik initialValues={initialValues} onSubmit={submit}>
+        {formik => (
+          <Form onSubmit={formik.handleSubmit}>
+            {/* A hack to disable automatic focus from Formik on the tags input.  */}
+            {/* eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex */}
```

Comment:
I wasn't aware that Formik could focus on an arbitrary element. Could you explain what the bug was looking like?

Reply:
The autocomplete input was automatically opened when opening the modal, covering up the "Popular tags" options:
<img width="610" alt="image" src="https://user-images.githubusercontent.com/1552194/218107794-7ec01587-13e0-4896-9418-355ae15e7043.png">

Reply:
I see; I suppose `focus-trap-react` is the one triggering the focus then. Your approach to fix this looks right, I'd just clarify this in the comments:

```suggestion
            {/* A hack to disable automatic focus from focus-trap-react on the tags input */}
            {/* eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex */}
```